sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java

```diff
@@ -0,0 +1,363 @@
+   }
+   
+   /** remove diversion-counter member
+    */
+   public void removeCounter() {
+       parameters.delete(ParameterNames.COUNTER);
+   }
+   
+
+   /**
+    * Gets the diversion-privacy parameter from the address param list.
+    * @return field
+    */
+   public String getPrivacy() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.PRIVACY);
+   }
+   
+   /**
+    * Sets the diversion-privacy parameter of the Diversion header.
+    *
+    * @param reason diversion-privacy parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setPrivacy(String p) throws ParseException {
+   	  if (p == null)
+             throw new NullPointerException("no privacy param");
+         else if (p.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
+         this.setParameter(ParameterNames.PRIVACY, p);
+   }
+   
+   /** Boolean function
+    * @return true if the parameter exist
+    */
+   public boolean hasPrivacy() {
+       return hasParameter(ParameterNames.PRIVACY);
+   }
+   
+   /** remove diversion-privacy member
+    */
+   public void removePrivacy() {
+       parameters.delete(ParameterNames.PRIVACY);
+   }
+   
+   
+   
+   /**
+    * Gets the diversion-screen parameter from the address param list.
+    * @return field
+    */
+   public String getScreen() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.SCREEN);
+   }
+
+   /**
+    * Sets the diversion-screen parameter of the Diversion header.
+    *
+    * @param reason diversion-privacy parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setScreen(String s) throws ParseException {
+       if (s == null) {
+           throw new NullPointerException("no screen param");
+       } else if (s.trim().equals("")) {
+           throw new ParseException("bad parameter", 0);
+       }
+       this.setParameter(ParameterNames.SCREEN, s);
+   }
+   
+   /** Boolean function
+    * @return true if the Tag exist
+    */
+   public boolean hasScreen() {
+       return hasParameter(ParameterNames.SCREEN);
+   }
+   
+   /** remove diversion-screen member
+    */
+   public void removeScreen() {
+       parameters.delete(ParameterNames.SCREEN);
+   }
+   
+   /**
+    * Gets the diversion-extension parameter from the address parameter list.
+    * @return field
+    */
+   public String getExtension() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.EXTENSION);
+   }
+   
+   /**
+    * Sets the diversion-extension parameter of the Diversion header.
+    *
+    * @param reason diversion-extension parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setExtension(String ex) throws ParseException {
```

Comment:
I think the extension methods can be removed and use regular setParameter methods for those given the syntax of extensions 
diversion-extension = token ["=" (token | quoted-string)]

Reply:
done